src/post/aws.py

```diff
@@ -1,28 +1,129 @@
 """Aws S3 Bucket functionality (namely upload_file)"""
+from __future__ import annotations
+
+import os
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Any, Optional
 
 import boto3
 from boto3.resources.base import ServiceResource
 from boto3.s3.transfer import S3Transfer
 from botocore.client import BaseClient
+from dotenv import load_dotenv
 
 from src.logger import set_log
+from src.models.tables import SyncTable
 
 log = set_log(__name__)
 
 
+@dataclass
+class BucketFileObject:
+    """
+    Basic Structure describing a file's location and associated block number in AWS bucket
+    Our files are structured as `table_name/cow_XXXXXXX.json`
+    where XXXXXX is an increasing sequence of integers.
+    More precisely, we use last_sync_block_number to indicate our last collection of records.
+    """
+
+    path: str
+    name: str
+    block: Optional[int]
+
+    @classmethod
+    def from_key(cls, object_key: str) -> BucketFileObject:
+        """
+        Decompose the unique identifier `object_key` into
+        more meaningful parts from which it can be reconstructed
+        """
+        path, name = object_key.split("/")
+
+        try:
+            block = int(name.strip("cow_").strip(".json"))
+        except ValueError:
+            # File structure does not satisfy block indexing!
+            block = None
```

Comment:
For testing purposes, we introduce files into the bucket that do not satisfy this naming convention. However since this is not intended for the actual code, I this this is a very good idea and I will adapt the tests. Thanks.